backend/api/Areas/Project/Mapping/Search/ProjectPropertyMap.cs

```diff
@@ -0,0 +1,113 @@
+using Mapster;
+using Model = Pims.Api.Areas.Property.Models.Search;
+using Entity = Pims.Dal.Entities;
+using Pims.Dal.Entities;
+using System.Linq;
+using System;
+using System.Collections.Generic;
+using Pims.Api.Areas.Project.Models.Search;
+using Pims.Api.Models;
+
+namespace Pims.Api.Areas.Project.Mapping.Search
+{
+    public class ProjectPropertyMap : IRegister
+    {
+        public void Register(TypeAdapterConfig config)
+        {
+
+            config.NewConfig<ProjectProperty, ProjectPropertyModel>()
+                .Map(dest => dest.PropertyTypeId,
+                    src => src.Building != null ? PropertyTypes.Building : PropertyTypes.Land)
+                .Map(dest => dest.Id, src => src.BuildingId ?? src.ParcelId)
+                .Map(dest => dest.Classification,
+                    src => src.Building != null ? src.Building.Classification.Name : src.Parcel.Classification.Name)
+                .Map(dest => dest.Name, src => src.Building != null ? src.Building.Description : src.Parcel.Description)
+                .Map(dest => dest.Address,
+                    src => src.Building != null
+                        ? FormatAddress(src.Building.Address)
+                        : FormatAddress(src.Parcel.Address))
+                .Map(dest => dest.City,
+                    src => src.Building != null ? src.Building.Address.City.Name : src.Parcel.Address.City.Name)
+                .Map(dest => dest.Municipality, src => src.Building != null ? src.Building.Parcel.Municipality : src.Parcel.Municipality)
+                .Map(dest => dest.Assessed,
+                    src => src.Building != null
+                        ? GetBuildingAssessedValue(src.Building.Evaluations)
+                        : GetParcelAssessedValue(src.Parcel.Evaluations))
+                .Map(dest => dest.NetBook,
+                    src => src.Building != null
+                        ? GetBuildingNetValue(src.Building.Fiscals)
+                        : GetParcelNetValue(src.Parcel.Fiscals))
+                .Map(dest => dest.Estimated,
+                    src => src.Building != null
+                        ? GetBuildingEstimatedValue(src.Building.Fiscals)
+                        : GetParcelEstimatedValue(src.Parcel.Fiscals))
+
+                .Map(dest => dest.AgencyCode,
+                    src => src.Building != null ? src.Building.Agency.Parent.Code : src.Parcel.Agency.Parent.Code)
+                .Map(dest => dest.SubAgency,
+                    src => src.Building != null ? src.Building.Agency.Name : src.Parcel.Agency.Name)
+
+                .Map(dest => dest.LandArea, src => src.Building != null ? src.Building.Parcel.LandArea : src.Parcel.LandArea)
```

Comment:
Might need to check if `src.Building.Parcel` is null

Reply:
This is marked as resolved but I don't see a solution here.

Reply:
missed it